sdk/spring-sdk/src/main/scala/kalix/springsdk/impl/WebClientProviderHolder.scala

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kalix.springsdk.impl
+
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.ConcurrentMap
+
+import akka.actor.ActorSystem
+import akka.actor.ClassicActorSystemProvider
+import akka.actor.ExtendedActorSystem
+import akka.actor.Extension
+import akka.actor.ExtensionId
+import akka.actor.ExtensionIdProvider
+import akka.annotation.InternalApi
+import kalix.javasdk.impl.ProxyInfoHolder
+import kalix.springsdk.WebClientProvider
+import org.springframework.http.HttpHeaders
+import org.springframework.http.MediaType
+import org.springframework.web.reactive.function.client.ExchangeFilterFunction
+import org.springframework.web.reactive.function.client.ExchangeFilterFunctions
+import org.springframework.web.reactive.function.client.WebClient
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[springsdk] object WebClientProviderHolder
+    extends ExtensionId[WebClientProviderHolder]
+    with ExtensionIdProvider {
+  override def get(system: ActorSystem): WebClientProviderHolder = super.get(system)
+
+  override def get(system: ClassicActorSystemProvider): WebClientProviderHolder = super.get(system)
+
+  override def createExtension(system: ExtendedActorSystem): WebClientProviderHolder =
+    new WebClientProviderHolder(system)
+  override def lookup: ExtensionId[_ <: Extension] = this
+}
+
+class WebClientProviderHolder(system: ExtendedActorSystem) extends Extension {
+  val webClientProvider = new WebClientProviderImpl(system)
```

Comment:
Needed to introduce one more Akka extension, for two reasons:

1. we need the actor system to pass the the provider impl
2. we should had only one instance because we are not caching the clients
3. we must control where the `WebClientProvider` can be inject (see KalixServer)

Point 3 is actually about having a safe way of building a singleton provider and inject it into Actions.